crates/bevy_ecs/src/system/query.rs

```diff
@@ -425,14 +426,12 @@ impl<'w, 's, Q: WorldQuery, F: ReadOnlyWorldQuery> Query<'w, 's, Q, F> {
     /// - [`for_each`](Self::for_each) for the closure based alternative.
     #[inline]
     pub fn iter(&self) -> QueryIter<'_, 's, Q::ReadOnly, F::ReadOnly> {
+        let state = self.state.as_readonly();
```

Comment:
I think that this set of changes is a bit of a readability regression, and makes the diff a lot noisier.

Reply:
That is a fair critique. My goal with those changes was to minimise the amount of operations happening within an `unsafe` block. I'll revert those changes because they aren't really relevant to the rest of the PR.